pages/governance/proposal/[proposalId].governance.tsx

```diff
@@ -439,23 +443,60 @@ export default function ProposalPage({
                     captionVariant="description"
                   >
                     <Box sx={{ textAlign: 'right' }}>
-                      <Typography>
-                        ~ {dayjs.unix(proposal.startTimestamp).format('DD MMM YYYY, hh:mm a')}
-                      </Typography>
+                      <Typography>{formatTime(proposal.startTimestamp)}</Typography>
                       <Typography variant="caption" color="text.muted">
                         {proposal.startBlock}
                       </Typography>
                     </Box>
                   </Row>
-                  {proposal.executed && (
+                  {proposalHasExpired && (
+                    <Row
+                      caption={
+                        <>
+                          <Trans>Ended</Trans>
+                          <Typography variant="caption" color="text.muted">
+                            Block
+                          </Typography>
+                        </>
+                      }
+                      sx={{ height: 48 }}
+                      captionVariant="description"
+                    >
+                      <Box sx={{ textAlign: 'right' }}>
+                        <Typography>{formatTime(proposal.expirationTimestamp)}</Typography>
+                        <Typography variant="caption" color="text.muted">
+                          {proposal.endBlock}
+                        </Typography>
+                      </Box>
+                    </Row>
+                  )}
+                  {!proposalHasExpired ? (
+                    <Row
+                      caption={
+                        <>
+                          <Trans>Ends</Trans>
+                          <Typography variant="caption" color="text.muted">
+                            Block
+                          </Typography>
+                        </>
+                      }
+                      sx={{ height: 48 }}
+                      captionVariant="description"
+                    >
+                      <Box sx={{ textAlign: 'right' }}>
+                        <Typography>{formatTime(proposal.expirationTimestamp)}</Typography>
+                        <Typography variant="caption" color="text.muted">
+                          {proposal.endBlock}
+                        </Typography>
+                      </Box>
+                    </Row>
+                  ) : (
                     <Row
                       caption={<Trans>Executed</Trans>}
                       sx={{ height: 48 }}
                       captionVariant="description"
                     >
-                      <Typography>
-                        {dayjs.unix(proposal.executionTime).format('DD MMM YYYY, hh:mm a')}
-                      </Typography>
+                      <Typography>{formatTime(proposal.executionTime)}</Typography>
```

Comment:
Lines 452-499 here can be simplified into a single ternary.

Reply:
I actually kind of prefer to have explicit statements instead of ternary for larger blocks so I think this is fine

Reply:
Whoops nvm didn't see this was changed